Library/Homebrew/extend/os/linux/software_spec.rb

```diff
@@ -1,6 +1,40 @@
 # typed: strict
 # frozen_string_literal: true
 
+class SoftwareSpec
+  extend T::Sig
+
+  GNU_GCC_REGEXP = T.let(/^(gcc)$|^(gcc@(4\.9|[5-9]|[10-99]))$/.freeze, Regexp)
+
+  sig { params(spec: T.any(String, T::Hash[T.untyped, T.untyped])).void }
+  def remove_gnu_compilers(spec)
+    spec = spec.keys.first if spec.is_a?(Hash)
+    return unless spec.is_a?(String)
+    return unless spec.match?(GNU_GCC_REGEXP)
+
+    # When a gcc compiler is defined, remove the other
+    # compilers to avoid having to define fails_with
+    # manually for all the others
+    gcc_version_to_keep = Formulary.factory(spec).version.to_s.slice(/\d+/)
+
+    versions = CompilerConstants::GNU_GCC_VERSIONS
+
+    if spec == "gcc"
+      # Special case for gcc: in linuxbrew-core, gcc is currently at version 5,
+      # so when we use 'depend_on "gcc"' we do not want to remove version 5.
+      # Once https://github.com/Homebrew/linuxbrew-core/pull/21380 is merged,
+      # this line becomes useless and we can remove the whole if/else logic
+      versions -= [gcc_version_to_keep] if gcc_version_to_keep != "5"
```

Comment:
I'd suggest sharing the `5` here with something used in the formula name in `Library/Homebrew/extend/os/linux/compilers.rb`

Reply:
Not really: the `5` here is coming from the brewed gcc formula in linuxbrew-core; not from the preferred compiler. It happens to be 5 for both, but from a code logic I think it's not really the same.

This if/else is temporary and I will delete it once https://github.com/Homebrew/linuxbrew-core/pull/21380 is merged (really soon), as I am saying in the comment. Only the "else" part will be kept, which is "easy" to understand (`versions -= [gcc_version_to_keep]`), at least I hope.

Reply:
Gotcha, fine with me to leave as-is then 👍🏻